### DISCUSSION TITLE
Installing NR, Mosquitto, Zigbee2mqtt on RPI zero 2w, ZigBee2MQTT constantly rebooting [SOLVED]

### DISCUSSION THREAD

**Discussion by CarloJongen** [Koenkk/zigbee2mqtt#21745] [General]
I have been trying to install NR, Mosquitto, Zigbee2mqtt on a RPI zero 2w for several days now and I'm getting desperate failing over and over again. I'm not new to this procedure because I have successfully installed these applications on several RPI4's in the past ( like a year ago).
I'm using a RPI Zero 2W in an expansion board (  [https://geekworm.com/products/m200](https://geekworm.com/products/m200) ) to install NR, Mosquitto and Zigbee2mqtt using the steps A, B and C below ( same steps I've used in the past ).

STEP A) [https://nodered.org/docs/getting-started/raspberrypi](https://nodered.org/docs/getting-started/raspberrypi)
STEP B) [https://randomnerdtutorials.com/how-to-install-mosquitto-broker-on-raspberry-pi/](https://randomnerdtutorials.com/how-to-install-mosquitto-broker-on-raspberry-pi/)
STEP C) [https://www.zigbee2mqtt.io/guide/installation/01_linux.html](https://www.zigbee2mqtt.io/guide/installation/01_linux.html)

I tried with both the OS Lite No Desktop 64 and the 32 bit version. Tried with different RPI zero 2w and without the expansion board. Several clean installs later. It always comes down to getting stuck on step C ( installing Zigbee2mqtt ). I'm getting stuck in step C when using the command "npm ci" or the "npm ci  --maxsockets 1". The installation freezes ( see attached files ) and the RPI gets warm. I can't even access the RPI via another ssh session. I need to hard reboot the RPI. As A Result, I can't use the "npm run build" command. I also tried different Zigbee2mqtt dongles ( see picture attached ) for that matter but with no result. Conclusion NR and Mosquitto works fine on the rpi zero 2w but the installation of Zigbee2mqtt fails.

Has anybody here successfully setup NR, Mosquitto, Zigbee2mqtt on a RPI zero 2w? 
How can I debug this issue?

![npm_ci_install_hangs](https://github.com/Koenkk/zigbee2mqtt/assets/162889511/030f3449-4a81-4c5e-9e29-c0315d0176d6)
![npm_ci_install_hangs_b](https://github.com/Koenkk/zigbee2mqtt/assets/162889511/f51ce5cd-0b2d-4bdb-987b-78785a48272e)
![npm_ci_install_hangs_c](https://github.com/Koenkk/zigbee2mqtt/assets/162889511/31ccb035-fb9b-44b9-a3df-f09b65c54f84)
![different_sniffers](https://github.com/Koenkk/zigbee2mqtt/assets/162889511/95925869-51c1-4c7e-a6e7-d965f3448faf)

**Comment by EvertDekker** [comment]
You are running the **npm run build** in the wrong directory, should be run in **/opt/zigbee2mqtt**
Just my 2 cents

**Reply by CarloJongen** [reply]
Eugh.. yes tunnel vision, thanks a lot Evert, this got me a step further. Now I'm wondering... The fact that the rpi hung at the step "npm ci". Does that mean it couldn't install all the things needed to run zigbee2mqqtt correctly?

**Comment by CarloJongen** [comment]
Continuing with the installation at [starting-zigbee2mqtt](https://www.zigbee2mqtt.io/guide/installation/01_linux.html#starting-zigbee2mqtt)
I get the Zigbee2mqtt to work but I get a lot of drop outs. Meaning Zigbee is not responding from time to time. Seams like zigbee2mqtt goes in a time out.
As a test I used another rpi zero 2w that sends every 5sec an mqtt get "{"state":""}" payload to  a [Smart Relay](https://www.zigbee2mqtt.io/devices/TO-Q-SY1-JZT.html). Everytime the Smart Relay doesn't respond it prints out a yellow line "date and time ZigBee2Mqtt! TIME OUT" see screen capture below. So Zigbee is working fine in a way but it's dropping out ( rebooting ) every minute or two.

What would be the reason for these time outs?

![zigbee2mqtt_after reboot](https://github.com/Koenkk/zigbee2mqtt/assets/162889511/5e3ac1dd-846b-4f6c-9433-aae73f394a87)

**Comment by CarloJongen** [comment]
I forgot to mention that the post below didn't help me, so I'm starting to believe that this issue is not completely resolved yet.
[ZigBee2MQTT service continuously restarting #21463](https://github.com/Koenkk/zigbee2mqtt/issues/21463)
@Koenkk

**Comment by CarloJongen** [comment]
Following the discussion below has solved my issue:
[discussion #21195](https://github.com/Koenkk/zigbee2mqtt/discussions/21195)

It came down to change the ZIGBEE2MQTT.SERVICE file into the lines below. The original ZIGBEE2MQTT.SERVICE file created with the installation procedure failed ZigBee2MQTT to be stable making it impossible to work with.

```
[Unit]
Description=zigbee2mqtt
After=network.target

[Service]
Environment=NODE_ENV=production
Type=exec
ExecStart=/usr/bin/npm start
WorkingDirectory=/opt/zigbee2mqtt
'#StandardOutput=inherit
StandardOutput=null
'# Or use StandardOutput=null if you don't want Zigbee2MQTT messages filling syslog, for more options see systemd.exec(5)
StandardError=inherit
Restart=always
RestartSec=10s
User=pi

[Install]
WantedBy=multi-user.target
```
